src/DBmysqlIterator.php

```diff
@@ -582,6 +582,14 @@ private function analyseCriterion($value)
                     $comparison = $value[0];
                     $criterion_value = $value[1];
                 } else {
+                    if (count($value) === 2 && isset($value[0], $value[1]) && is_array($value[1])) {
+                        if ($value[0] === 'IN') {
```

Comment:
IN and NOT IN are treated as special operators in criteria now to help with handling dynamic criteria. The main requirement was being able to have a NOT IN criteria without needing to wrap anything in a NOT property. Support for the IN operator was just added to stay consistent.

Reply:
If Cédric's proposition is OK; it's way easy to read :)